packages/story-editor/src/components/library/panes/media/local/karma/mediaFetching.karma.js

```diff
@@ -49,11 +49,12 @@ describe('MediaPane fetching', () => {
     const initialElements =
       within(mediaGallery).queryAllByTestId(/^mediaElement-/);
 
+    // ensure fixture.screen has loaded before calling expect to prevent immediate failure
+    if (initialElements.length !== LOCAL_MEDIA_PER_PAGE) {
+      await fixture.events.sleep(500);
+    }
+
```

Comment:
instead of sleeping here we could use `waitFor`. If that time is less than 500 ms then this won't fail, but this will also pass if it takes longer than 500 ms.
```suggestion
    await waitFor(() => {
      if (initialElements.length !== LOCAL_MEDIA_PER_PAGE) {
        throw new Error('Not ready - initial media elements not loaded yet');
      }
    });

```

Reply:
That is what we are already doing. So when the if statement is true the error gets thrown resulting in a failed test.

Reply:
At least I think that would be essentially the same thing as we are currently doing.